### ISSUE TITLE
Bug: exception when copyToken used with some updates

### ISSUE THREAD

**Issue by Merudo** [RPTools/maptool#814]
**Describe the bug**
Using `copyToken` with the following updates raises an exception: x, y, layer, tokenImage, portraitImage, handoutImage

**To Reproduce**
1. Run MapTool in developer mode (console)
2. Create a map with a `Dragon` token on Grasslands
3. Run the following code:
```
[h: updates = json.set("{}", "x", 1)]
[h: copyToken("Dragon", 1, "Grasslands", updates)]
```
4. Check the console. An exception is thrown.
 
**Expected behavior**
No exceptions.

**MapTool Info**
- Version: 1.5.6 Develop
- Install: New

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: 10

**Additional context**
Exception is thrown because some code in `copyToken` is trying to send a token update to the server before the token is even created on the server.

**Comment by Merudo** [comment]
PR #815 solves this.

**Comment by Phergus** [comment]
Good find.

**Comment by Phergus** [comment]
PR #815 fixed most but not all.

Additional steps to above:
1. Copy the top-down "Dragon" _token_ from Default -> Tokens to the map.
2. Add a handout image to the new token. (It already has a portrait.)
3. Name it "Dragon 1" if it didn't default to that.

Run this code:
```
[h: updates = json.set("{}", "x", 1, "y", 2)]
[h,token("Dragon 1"): id = getTokenImage()]
[h: updates = json.set(updates, "tokenImage", id)]
[h,token("Dragon 1"): id2 = getTokenPortrait()]
[h: updates = json.set(updates, "tokenPortrait", id2)]
[h,token("Dragon 1"): id3 = getTokenHandout()]
[h: updates = json.set(updates, "tokenHandout", id3)]
<pre>[r: json.indent(updates,2)]</pre>
[h: copyToken("Dragon", 1, "Clone vat", updates)]
```
Only the token image is changed.  No change to portrait or handout.

Try this code:
```
[h: updates = json.set("{}", "x", 1, "y", 2)]
[h: updates = json.set(updates, "tokenImage", "Dragon 1")]
[h: updates = json.set(updates, "tokenPortrait", "Dragon 1")]
[h: updates = json.set(updates, "tokenHandout", "Dragon 1")]
<pre>[r: json.indent(updates,2)]</pre>
[h: copyToken("Dragon", 1, "Clone vat", updates)]
```
Macro fails with:
`Argument number 1 invalid argument type for function "setImage"`

**Comment by Phergus** [comment]
PR #815 fixes the exceptions but this has revealed other bugs that were already there in 1.5.6.
* `imageToken` option of `update` only works with `Asset IDs` and not with token name.
* `imagePortrait` and `imageHandout` don't work with either.

Add additional steps to reported steps:
1. Copy the top-down "Dragon" token from Default -> Tokens to the map.
2. Add a handout image to the new token. (It already has a portrait.)
3. Name it "Dragon 1" if it didn't default to that.

Update with asset IDs.
```
[h: updates = json.set("{}", "x", 1, "y", 2)]
[h,token("Dragon 1"): id = getTokenImage()]
[h: updates = json.set(updates, "tokenImage", id)]
[h,token("Dragon 1"): id2 = getTokenPortrait()]
[h: updates = json.set(updates, "tokenPortrait", id2)]
[h,token("Dragon 1"): id3 = getTokenHandout()]
[h: updates = json.set(updates, "tokenHandout", id3)]
<pre>[r: json.indent(updates,2)]</pre>
[h: copyToken("Dragon", 1, "Clone vat", updates)]
```
Only token image is changed.

Update with token names
```
[h: updates = json.set("{}", "x", 1, "y", 2)]
[h: updates = json.set(updates, "tokenImage", "Dragon 1")]
[h: updates = json.set(updates, "tokenPortrait", "Dragon 1")]
[h: updates = json.set(updates, "tokenHandout", "Dragon 1")]
<pre>[r: json.indent(updates,2)]</pre>
[h: copyToken("Dragon", 1, "Clone vat", updates)]
```
Returns error:
`Argument number 1 invalid argument type for function "setImage".`

**Comment by Merudo** [comment]
Instead of `tokenPortrait` and `tokenHandout` the proper update fields are `portraitImage` & `handoutImage`.

**Comment by Merudo** [comment]
PR #816 changes `portraitImage` and `handoutImage` to  `tokenPortrait` and `tokenHandout`.

**Comment by Phergus** [comment]
Weird.  I thought I used the keywords from your post.  That's what happens when you don't read the warnings on the cold medicine:  _Don't operate heavy machinery or write macro code while using._

> Also, tokenImage / portraitImage / handoutImage do work with image tokens, just not regular tokens. Are they suppose to work for both?

Probably not.  Just my drug fogged brain not reading things correctly.  Don't know that there is a real need for that to work.  At the time I wrote the macro I just made the assumption that it would grab the right image.
